main.js

```diff
@@ -129,25 +144,53 @@ function handleTitleClick(event) {
   });
   removePosts();
 
-  const allPostsContainer = document.getElementById("all-posts-container");
   currentPost.render(allPostsContainer, false);
 
   const link = document.getElementById("back");
   link.classList.remove("hidden");
   link.addEventListener("click", handleBackToAllPosts);
 }
+
 function removePosts() {
   const posts = document.querySelectorAll(".all-posts-page .post-container");
   posts.forEach(function(post) {
     post.remove();
   });
 }
 
+function getAuthorClickHandlerForPost(post) {
+  return function handleAuthorCick(event) {
+    const path = event.target.dataset.path;
+    const postsByAuthor = allPosts.filter(function(post) {
+      return path == post.author;
```

Comment:
let's always use strict equality in our JS since `==` and `!=` can lead to lots of confusing mistakes. add the `eqeqeq` to the eslint config and fix these https://eslint.org/docs/rules/eqeqeq